src/devtools/client/debugger/src/reducers/pause.ts

```diff
@@ -252,6 +270,10 @@ export function getPauseHistory(state: UIState) {
 export function getPauseHistoryIndex(state: UIState) {
   return state.pause.pauseHistoryIndex;
 }
+// for components that should not switch to a new point before its `pauseId` is available
+export function getMostRecentPause(state: UIState): PauseHistoryData | undefined {
+  return state.pause.pauseHistory[state.pause.pauseHistoryIndex];
+}
```

Comment:
I'd like to better understand this specific change. Maybe we can chat about it during sync today (or now if you're free)?

I'm a bit concerned about us relying on a selector like this because it seems easy to overlook/forget. It's also not obvious to me when or why a component would want to consume the _current_ execution point but the _previous_ pause. Seems like those would need to be kept in sync.

So looking at RDT specifically, maybe we should have a `prevRef` that stores both the previous (non-null?) execution point and pause id (so the two are always in sync, internally)? am I overlooking something?